src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackExecutor.cs

```diff
@@ -0,0 +1,452 @@
+
+        }
+
+        /// <summary>
+        /// Executes the HPL workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            DateTime startTime = DateTime.UtcNow;
+            await this.ExecuteCommandAsync("make", $"arch=Linux_GCC", this.HPLDirectory, telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+
+            this.SetParameters();
+            await this.ConfigureHPLdatFileAsync(telemetryContext, cancellationToken).ConfigureAwait(false);
+
+            string results;
+            if (this.CpuArchitecture == Architecture.X64)
+            {
+                results = await this.ExecuteElevatedCommandAsync("runuser", $"-u azureuser -- mpirun --use-hwthread-cpus -np {Environment.ProcessorCount} ./xhpl", this.PlatformSpecifics.Combine(this.HPLDirectory, "bin", "Linux_GCC"), telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+            }
+            else
+            {
+                results = await this.ExecuteElevatedCommandAsync("runuser", $"-u azureuser -- mpirun -np {Environment.ProcessorCount} ./xhpl", this.PlatformSpecifics.Combine(this.HPLDirectory, "bin", "Linux_GCC"), telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+            }
+
+            DateTime endTime = DateTime.UtcNow;
+            this.LogMetrics(results, startTime, endTime, telemetryContext, cancellationToken);
+        }
+
+        private void SetParameters()
+        {
+            if (this.N == null)
+            {
+                this.N = (Environment.ProcessorCount * 10000).ToString();
+            }
+
+            // gives you P*Q = Environment.ProcessorCount and P <= Q and  Q-P to be the minimum possible value.
+            this.ProcessRows = 1;
+            this.ProcessColumns = Environment.ProcessorCount;
+            for (int i = 2; i <= Math.Sqrt(Environment.ProcessorCount); i++)
+            {
+                if (Environment.ProcessorCount % i == 0)
+                {
+                    int j = Environment.ProcessorCount / i;
+                    if (j - i < this.ProcessColumns - this.ProcessRows)
+                    {
+                        this.ProcessRows = i;
+                        this.ProcessColumns = j;
+                    }
+                }
+            }
+
+        }
+
+        private void ThrowIfPlatformIsNotSupported()
+        {
+            if (this.Platform != PlatformID.Unix)
+            {
+                throw new WorkloadException(
+                    $"The HPL workload is currently only supported on the following platform/architectures: " +
+                    $"'{PlatformSpecifics.GetPlatformArchitectureName(PlatformID.Unix, Architecture.X64)}', '{PlatformSpecifics.GetPlatformArchitectureName(PlatformID.Unix, Architecture.Arm64)}'. ",
```

Comment:
Do me a favor and let's add the following public static properties to the PlatformSpecifics class. I keep seeing a need for the platform architecture names in error messages. This will allow us to shorten the length of some of these error messages and we can use string.Join() techniques for ease-of-coding as well.

class PlatformSpecifics
{
      public static readonly LinuxX64 = PlatformSpecifics.GetPlatformArchitectureName(PlatformID.Unix, ....);
      public static readonly LinuxArm64 = PlatformSpecifics.GetPlatformArchitectureName(PlatformID.Unix, ....);
      public static readonly WinX64 = PlatformSpecifics.GetPlatformArchitectureName(PlatformID.Win32NT, ....);
      public static readonly WinArm64 = PlatformSpecifics.GetPlatformArchitectureName(PlatformID.Win32NT, ....);
}